packages/aws-cdk-lib/aws-ec2/lib/nat.ts

```diff
@@ -399,6 +419,115 @@ class PrefSet<A> {
   }
 }
 
+/**
+ * Modern NAT provider which uses NAT Instances.
+ * The instance uses Amazon Linux 2 as the operating system.
+ */
+export class NatInstanceProviderAL2 extends NatProvider implements IConnectable {
+  private gateways: PrefSet<Instance> = new PrefSet<Instance>();
+  private _securityGroup?: ISecurityGroup;
+  private _connections?: Connections;
+
+  constructor(private readonly props: NatInstanceProps) {
+    super();
+
+    if (props.defaultAllowedTraffic !== undefined && props.allowAllTraffic !== undefined) {
+      throw new Error('Can not specify both of \'defaultAllowedTraffic\' and \'defaultAllowedTraffic\'; prefer \'defaultAllowedTraffic\'');
+    }
+
+    if (props.keyName && props.keyPair) {
+      throw new Error('Cannot specify both of \'keyName\' and \'keyPair\'; prefer \'keyPair\'');
+    }
+  }
+
+  public configureNat(options: ConfigureNatOptions) {
+    const defaultDirection = this.props.defaultAllowedTraffic ??
+      (this.props.allowAllTraffic ?? true ? NatTrafficDirection.INBOUND_AND_OUTBOUND : NatTrafficDirection.OUTBOUND_ONLY);
+
+    // Create the NAT instances. They can share a security group and a Role.
+    const machineImage = this.props.machineImage || new AmazonLinuxImage({ generation: AmazonLinuxGeneration.AMAZON_LINUX_2 });
```

Comment:
It'll be more useful if it automatically sets an AMI [cpu type](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.AmazonLinuxImageProps.html#cputype) according to the instance type [architecture](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.InstanceType.html#architecture). 

I believe lots of people will use t4g.nano because it's the cheapest instance type.

Reply:
I don't want to make the default value based on assumption. I haven't found any support evidence that this is the most wanted instance type, so I'll keep the instance type as is.

Reply:
Thanks. I didn't mean make arm architecture default. Instead we can infer the appropriate architecture from an instance type passed from a user. This way users will not need to be aware of arch type of the machine image. Currently if a user choose a graviton instance type, it fails during CFn deployment because of architecture mismatch, resulting in bad UX.